src/lib/spinel/radio_spinel.cpp

```diff
@@ -263,19 +187,17 @@ otError RadioSpinel::CheckSpinelVersion(void)
     return error;
 }
 
-bool RadioSpinel::IsRcp(bool &aSupportsRcpApiVersion, bool &aSupportsRcpMinHostApiVersion)
+void RadioSpinel::InitializeCaps(bool &aSupportsRcpApiVersion, bool &aSupportsRcpMinHostApiVersion)
 {
-    uint8_t        capsBuffer[kCapsBufferSize];
-    const uint8_t *capsData         = capsBuffer;
-    spinel_size_t  capsLength       = sizeof(capsBuffer);
+    const uint8_t *capsData;
+    spinel_size_t  capsLength;
     bool           supportsRawRadio = false;
-    bool           isRcp            = false;
+
+    capsData = mSpinelDriver->GetCapsBuffer(capsLength);
```

Comment:
I think since the returned pointer must points to the array member `mCapsBuffer` which will never be a nullptr it's not quite necessary to check this. Thoughts?

Reply:
The method has been removed.